BaseCollector.py

```diff
@@ -107,11 +111,15 @@ def get_vms_by_target(self):
         return self.target_vms
 
     def get_project_ids_by_target(self):
-        token = self.get_target_tokens()
-        token = token[self.target]
-        uuids = self.get_vms_by_target()
-        project_ids = Vrops.get_project_ids(self.target, token, uuids, self.collector)
-        return project_ids
+        try:
+            token = self.get_target_tokens()
+            token = token[self.target]
```

Comment:
Wouldn't this still fail, if `get_target_tokens` is returning `{}`?
Latest `get_project_ids` will fail because of the missing token and therefore the `ConnectionError` appears. As we now this is going to happen, avoid it.

Reply:
No, it wouldn't fail. Because of line 17 `while` os.environ['TARGET'] not in self.get_target_tokens():`
The target will not be in the empty dictionary and thus will retry in 30 minutes. Or is restored earlier, as we have already observed.

Reply:
And to the second comment: No, every collector is now checking the token first. The code won't reach `get_project_ids`.